src/lib.rs

```diff
@@ -476,6 +512,17 @@ mod tests {
         assert_eq!(source().to_string(), my_err.source().unwrap().to_string());
     }
 
+    #[test]
+    #[allow(dead_code)]
+    fn struct_with_source_and_others() {
+        use std::{io, error::Error};
+        custom_error!(MyError{x:u8, source:io::Error}="{x}");
+        fn source() -> io::Error { io::ErrorKind::AlreadyExists.into() };
```

Comment:
You could simply define the error and then `.copy()` it...

Reply:
Not sure if I get you wrong, the error doesn't implement clone.

Reply:
Sorry, never mind :/

Reply:
No problem. I stole that from your previous test. I also thought that it looked weird at first.